go/common/tribool/tribool.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package tribool
+
+// Tribool is a three-valued logic type.
+// It can be either True, False or Unknown.
+// It is useful for representing the result of a comparison
+// when the comparison is not possible or could not be determined.
+type Tribool struct {
+	value byte
+}
+
+// Unknown returns true if the value is unknown.
+func (t Tribool) Unknown() bool {
+	return t.value > 1
+}
+
+// False returns true if the value is false.
+func (t Tribool) False() bool {
+	return t.value == 0
+}
+
+// True returns true if the value is true.
+func (t Tribool) True() bool {
+	return t.value == 1
+}
+
+// String returns a string representation of the Tribool.
+func (t Tribool) String() string {
+	switch t.value {
+	case 0:
+		return "False"
```

Comment:
Should we align with the `bool` type and write `false`, `true`, and `unknown` with a lower-case first letter?

Reply:
yes, I think it is a good idea